server/src/main/java/org/opensearch/index/recovery/RemoteStoreRestoreService.java

```diff
@@ -150,4 +201,68 @@ public void clusterStateProcessed(String source, ClusterState oldState, ClusterS
         });
 
     }
+
+    private void validate(
+        ClusterState currentState,
+        Map<String, Tuple<Boolean, IndexMetadata>> indexMetadataMap,
+        boolean restoreAllShards
+    ) {
+        String errorMsg = "cannot restore index [%s] because an open index with same name already exists in the cluster.";
+        for (Map.Entry<String, Tuple<Boolean, IndexMetadata>> indexMetadataEntry : indexMetadataMap.entrySet()) {
+            String indexName = indexMetadataEntry.getKey();
+            IndexMetadata indexMetadata = indexMetadataEntry.getValue().v2();
+            String indexUuid = indexMetadata.getIndexUUID();
+            boolean fromRemoteStore = indexMetadataEntry.getValue().v1();
+            if (indexMetadata.getSettings().getAsBoolean(SETTING_REMOTE_STORE_ENABLED, false)) {
+                if (restoreAllShards && IndexMetadata.State.CLOSE.equals(indexMetadata.getState()) == false) {
+                    throw new IllegalStateException(String.format(Locale.ROOT, errorMsg, indexName) + " Close the existing index.");
+                }
+
+                if (fromRemoteStore) {
```

Comment:
nit: Name is not providing much details here. For this API, data anyways come from remote store. As I understand, what we want to check is if `metadataFromRemoteStore`

Reply:
Sure 👍

Reply:
had already done this but got lost in rebases. will update